src/compiler/declarationEmitter.ts

```diff
@@ -396,6 +398,12 @@ namespace ts {
                 }
             }
 
+            function emitTypePredicate(type: TypePredicateNode) {
+                writeEntityName(type.parameterName);

```

Comment:
I looked at this, emitEntityName checks the visibility of the entity name, which is primarily for testing type visibility and does not understand that the parameterName is in the value space. As a result, you get errors that the signature references a private type.